src/osa/scripts/gain_selection.py

```diff
@@ -134,19 +260,26 @@ def apply_gain_selection(date: str, start: int, end: int, output_basedir: Path =
         else:
             tool = "lstchain_r0_to_r0g"
 
-    run_summary_dir = Path("/fefs/aswg/data/real/monitoring/RunSummary")
+    run_summary_dir = Path(cfg.get("LST1", "RUN_SUMMARY_DIR"))
     run_summary_file = run_summary_dir / f"RunSummary_{date}.ecsv"
     summary_table = Table.read(run_summary_file)
+
+    if len(summary_table) == 0:
+        log.warning(f"No runs are found in the run summary of {date}. Nothing to do. Exiting.")
+        sys.exit(0)
+
     # Apply gain selection only to DATA runs
     data_runs = summary_table[summary_table["run_type"] == "DATA"]
     log.info(f"Found {len(data_runs)} DATA runs to which apply the gain selection")
 
-    output_dir = output_basedir / date
-    log_dir = output_basedir / "log" / date
-    output_dir.mkdir(parents=True, exist_ok=True)
-    log_dir.mkdir(parents=True, exist_ok=True)
+    base_dir = Path(cfg.get("LST1", "BASE"))
+    r0_dir = base_dir / "R0" / date
+    output_dir = base_dir / f"R0G/{date}"
+    log_dir = base_dir / f"R0G/log/{date}"
     log_file = log_dir / f"r0_to_r0g_{date}.log"
-    r0_dir = Path(f"/fefs/aswg/data/real/R0/{date}")
+    if not simulate:
+        output_dir.mkdir(parents=True, exist_ok=True)
```

Comment:
although it is more clear with this line, it is not needed. The line below `log_dir` will create the parent output_dir directory

Reply:
I have just noticed now that the line 281 (output_dir.mkdir(parents=True, exist_ok=True)) is necessary, since output_dir is not a parent directory of log_dir.